adapters/p2p2consensus/proposal.go

```diff
@@ -6,69 +6,137 @@ import (
 	consensus "github.com/NethermindEth/juno/consensus/types"
 	"github.com/NethermindEth/juno/core"
 	"github.com/NethermindEth/juno/core/felt"
-	"github.com/NethermindEth/juno/utils"
 	p2pconsensus "github.com/starknet-io/starknet-p2pspecs/p2p/proto/consensus/consensus"
 )
 
-func AdaptProposalInit(msg *p2pconsensus.ProposalInit) consensus.ProposalInit {
+func AdaptProposalInit(msg *p2pconsensus.ProposalInit) (consensus.ProposalInit, error) {
+	if err := validateProposalInit(msg); err != nil {
+		return consensus.ProposalInit{}, err
+	}
+
+	var proposer felt.Felt
+	if err := proposer.SetBytesCanonical(msg.Proposer.Elements); err != nil {
+		return consensus.ProposalInit{}, err
+	}
+
 	return consensus.ProposalInit{
 		BlockNum: msg.BlockNumber,
-		Proposer: *new(felt.Felt).SetBytes(msg.Proposer.Elements),
-	}
+		Proposer: proposer,
+	}, nil
 }
 
-func AdaptBlockInfo(msg *p2pconsensus.BlockInfo) consensus.BlockInfo {
+func AdaptBlockInfo(msg *p2pconsensus.BlockInfo) (consensus.BlockInfo, error) {
+	if err := validateBlockInfo(msg); err != nil {
+		return consensus.BlockInfo{}, err
+	}
+
+	var builder felt.Felt
+	if err := builder.SetBytesCanonical(msg.Builder.Elements); err != nil {
+		return consensus.BlockInfo{}, err
+	}
+
 	return consensus.BlockInfo{
 		BlockNumber:       msg.BlockNumber,
-		Builder:           *new(felt.Felt).SetBytes(msg.Builder.Elements),
+		Builder:           builder,
 		Timestamp:         msg.Timestamp,
 		L2GasPriceFRI:     *p2p2core.AdaptUint128(msg.L2GasPriceFri),
 		L1GasPriceWEI:     *p2p2core.AdaptUint128(msg.L1DataGasPriceWei),
 		L1DataGasPriceWEI: *p2p2core.AdaptUint128(msg.L1DataGasPriceWei),
 		EthToStrkRate:     *p2p2core.AdaptUint128(msg.EthToStrkRate),
 		L1DAMode:          core.L1DAMode(msg.L1DaMode),
-	}
+	}, nil
 }
 
-func AdaptProposalCommitment(msg *p2pconsensus.ProposalCommitment) consensus.ProposalCommitment {
+func AdaptProposalCommitment(msg *p2pconsensus.ProposalCommitment) (consensus.ProposalCommitment, error) {
+	if err := validateProposalCommitment(msg); err != nil {
+		return consensus.ProposalCommitment{}, err
+	}
+
+	var builder felt.Felt
+	if err := builder.SetBytesCanonical(msg.Builder.Elements); err != nil {
```

Comment:
nit:
```suggestion
	if err := builder.SetBytesCanonical(msg.Builder.GetElements()); err != nil {
```
So we don't need a separate `validateProposalCommitment` function

Reply:
If we don't need this then I agree using `GetElements()` will simplify the code quite a bit